app/Exceptions/Handler.php

```diff
@@ -4,14 +4,23 @@
 
 namespace Brewmap\Exceptions;
 
-use Illuminate\Auth\AuthenticationException;
+use Brewmap\Exceptions\Auth\ApiException;
 use Illuminate\Foundation\Exceptions\Handler as LaravelHandler;
 use Symfony\Component\HttpFoundation\Response;
+use Throwable;
 
 class Handler extends LaravelHandler
 {
-    protected function unauthenticated($request, AuthenticationException $exception): Response
+    public function render($request, Throwable $exception): Response
     {
-        return response()->json(["message" => $exception->getMessage()], Response::HTTP_UNAUTHORIZED);
+        if ($request->isJson() || $exception instanceof ApiException) {
```

Comment:
How about implementing `render` method in `ApiException` rather than making `instanceof` checks in Handler for each of custom exception?

So base `ApiException` could look something like this:
```
class ApiException extends Exception
{
    protected $message = "Server error.";
    protected int $status = Response::HTTP_INTERNAL_SERVER_ERROR;

    public function render(): JsonResponse
    {
        return new JsonResponse(['message' => $this->message], $this->status);
    }
}
```
so no additional code in Handler will be needed

Reply:
https://laravel.com/docs/7.x/errors#renderable-exceptions

Reply:
Nice idea.

Reply:
Or not. I'm still wondering if exception should know how it should be rendered. I don't think it's exception's responsibility at all.

Reply:
I've changed ApiException to abstract class extending Exception for now.
We can discuss it again in #88